src/amberscript/parser_buffer_test.cc

```diff
@@ -1090,17 +1118,14 @@ TEST_F(AmberScriptParserTest, BufferMissingDataFilePng) {
 TEST_F(AmberScriptParserTest, BufferDataFilePng) {
   std::string in = "BUFFER my_buffer FORMAT R8G8B8A8_UNORM FILE PNG foo.png";
 
-  Parser parser;
+  DummyDelegate delegate;
+  Parser parser(&delegate);
   Result r = parser.Parse(in);
-  ASSERT_TRUE(r.IsSuccess()) << r.Error();
-
+  ASSERT_TRUE(r.IsSuccess());
```

Comment:
Can we update the `DummyDelegate` to store a flag when the load is called and verify that we call into `LoadBufferData` for each one of these?

Reply:
The member function is const, so I cannot store anything in the delegate itself. What I did was store the file type in buffer data.